internal/util/k8sutil/k8sutil.go

```diff
@@ -113,3 +115,15 @@ func GetTypeMetaFromBytes(b []byte) (t metav1.TypeMeta, err error) {
 		Kind:       u.GetKind(),
 	}, nil
 }
+
+// dns1123LabelRegexp defines the character set allowed in a DNS 1123 label.
+var dns1123LabelRegexp = regexp.MustCompile("[^a-zA-Z0-9]+")
+
+// FormatOperatorNameDNS1123 ensures name is DNS1123 label-compliant by
+// replacing all non-compliant UTF-8 characters with "-".
+func FormatOperatorNameDNS1123(name string) string {
```

Comment:
Have we not a func in the apimachinery that we could call directly instead of creating this one?
If not, could we add a comment to make clear that it is a copy and paste from there? WDYT?

Reply:
AFAIK there is no function in apimachinery to format a string as a DNS1123 label. Nothing is copied from apimachinery either.

Reply:
If there is no library to do this in the SDK or one of our main deps like controller-runtime or kubebuilder, then I'm okay with this function.